packages/contracts/src/contracts/current/protocol/Exchange/MixinSettlementProxy.sol

```diff
@@ -87,9 +91,10 @@ contract MixinSettlementProxy is
         if (order.feeRecipientAddress != address(0)) {
             if (order.makerFeeAmount > 0) {
                 makerFeeAmountPaid = getPartialAmount(takerTokenFilledAmount, order.takerTokenAmount, order.makerFeeAmount);
+
                 require(
                     TRANSFER_PROXY.transferFrom(
-                        ZRX_TOKEN,
+                        encodeERC20Metadata(ZRX_TOKEN),
```

Comment:
I'm don't think we want to hard code ERC20 encoding here. I could see us having different proxies that are compatible with the same tokens.

Reply:
Updated to pass the erc20 proxy id into constructor.